clients/gateway/gateway.go

```diff
@@ -96,28 +96,16 @@ func NewClient(gatewayURL string, log utils.SimpleLogger) *Client {
 }
 
 func (c *Client) AddTransaction(txn json.RawMessage) (json.RawMessage, error) {
-	endpoint := c.url + "/add_transaction"
-
-	body, err := c.post(endpoint, txn)
-	if err != nil {
-		return nil, err
-	}
-	defer body.Close()
-
-	res, readErr := io.ReadAll(body)
-	if readErr != nil {
-		return nil, readErr
-	}
-
-	return res, nil
+	return c.post(c.url+"/add_transaction", txn)
 }
 
 // post performs additional utility function over doPost method
-func (c *Client) post(url string, data any) (io.ReadCloser, error) {
+func (c *Client) post(url string, data any) ([]byte, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 
 	resp, err := c.doPost(ctx, url, data)
+	defer resp.Body.Close()
```

Comment:
if `doPost` returns an error will `resp` be nil ?

Reply:
:facepalm: thanks